server/NucleiSegmentation/NucleiSegmentation.py

```diff
@@ -17,15 +18,33 @@
 }
 
 
+def read_image(file):
+    return skimage.io.imread(file)[:, :, :3]
+
+
 def main(args):
 
     #
     # Read Input Image
     #
     print('>> Reading input image')
 
-    imInput = skimage.io.imread(args.inputImageFile)[:, :, :3]
-
+    imInput = None
+    imageFile = args.inputImageFile
+    if os.path.isdir(imageFile):
+        dir = imageFile
+        for file in os.listdir(dir):
```

Comment:
Currently, after selecting a file for the inputImageFile in your earlier front-end and submitting the job, girder_worker takes care of copying the file from the girder database to the worker machine, and set args.inputImageFile to a local file path on the worker machine.

If we set it to a directory, may be girder_worker will try to copy the whole directory/folder from Girder which might take more time if it contains many images.